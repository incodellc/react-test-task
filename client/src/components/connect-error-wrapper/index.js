import ConnectErrorWrapper from './ConnectErrorWrapper';
export default ConnectErrorWrapper;