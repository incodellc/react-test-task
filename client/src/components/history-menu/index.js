import HistoryMenu from './HistoryMenu';
export default HistoryMenu;