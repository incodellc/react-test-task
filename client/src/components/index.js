import InfoBar from './info-bar';
import Layout from './layout';
import ConnectErrorWrapper from './connect-error-wrapper';
import TickerList from './ticker-list';
import TickerListItem from './ticker-list-item';

export { InfoBar, Layout, ConnectErrorWrapper, TickerList, TickerListItem };