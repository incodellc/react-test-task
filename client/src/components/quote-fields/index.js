import QuoteFields from './QuoteFields';
export default QuoteFields;