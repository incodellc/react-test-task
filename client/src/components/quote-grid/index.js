import QuotesGrid from './QuotesGrid';
export default QuotesGrid;