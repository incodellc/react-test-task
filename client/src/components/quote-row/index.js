import QuoteRow from './QuoteRow';
export default QuoteRow;