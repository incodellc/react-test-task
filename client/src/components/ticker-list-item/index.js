import TickerListItem from './TickerListItem';
export default TickerListItem;