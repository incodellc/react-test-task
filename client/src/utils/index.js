import moment from 'moment';
export { quotesExample, tickers, fields, historyLengthOptions } from './presets';

export const rewriteObjectProps = (prevObject, newPropsValues) => {
    return {
        ...prevObject,
        ...newPropsValues
    };
};

export const sortByKey = (arrayOfObjects, sortParams) => {
    let { target, isDate, reverse } = sortParams;
    let data = arrayOfObjects.slice();

    if (target === null) {
        return data;
    }

    const compareByKey = (key, isDate = false, isDesc = false) => {
        let orderFlag = isDesc ? -1 : 1;
        return (a, b) => {
            let valA = typeof a[key] === 'string' && isNaN(a[key]) ? a[key].toUpperCase() : typeof +a[key] ===  'number' ? +a[key] : '';
            let valB = typeof b[key] === 'string' && isNaN(a[key]) ? b[key].toUpperCase() : typeof +b[key] ===  'number' ? +b[key] : '';
            if (isDate) {
                valA = valA === '' ? moment(+valA) : moment(valA, 'DD-MM-YYYY HH:mm');
                valB = valB === '' ? moment(+valB) : moment(valB, 'DD-MM-YYYY HH:mm');
            }
            let result = (valA < valB) ? -1 : (valA > valB) ? 1 : 0;
            return result * orderFlag;
        };
    };

    return data.sort(compareByKey(target, isDate, reverse));
};
