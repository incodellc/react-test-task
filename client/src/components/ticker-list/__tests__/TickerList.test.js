import React from 'react';
import { renderWithRedux } from '../../../utils/test-utils';
import { TickerList } from '../../../components';
import { cleanup } from '@testing-library/react';

describe('Check TickerList component', () => {
    afterEach(cleanup);

    it('component renders correctly', () => {
        const { asFragment } = renderWithRedux(<TickerList />);
        expect(asFragment()).toMatchSnapshot();
    });
    
    it('component should contain correct number of TickerListItem comp-s', () => {
        const { getByTestId, getAllByTestId, store } = renderWithRedux(<TickerList />);
        expect(getByTestId('ticker-list')).toContainElement(getAllByTestId('ticker-list-item')[0]);
        expect(getAllByTestId('ticker-list-item').length).toEqual(store.getState().tickers.length);
    });
});