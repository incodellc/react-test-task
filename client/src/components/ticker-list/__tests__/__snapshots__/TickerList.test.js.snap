// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check TickerList component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui celled selection middle aligned list"
    data-testid="ticker-list"
    role="list"
  >
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="AAPL"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-negative"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              213.41
            </div>
            <div
              class="label"
            >
              <span
                class="pricedown"
              >
                -38.72
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="red arrow down big icon"
            data-testid="ticker-list-item-icon-down"
          />
        </div>
        <div
          class="header"
        >
          AAPL
        </div>
        <div
          class="meta"
        >
          Apple Inc.
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="GOOGL"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-positive"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              253.86
            </div>
            <div
              class="label"
            >
              <span
                class="priceup"
              >
                +5.59
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="green arrow up big icon"
            data-testid="ticker-list-item-icon-up"
          />
        </div>
        <div
          class="header"
        >
          GOOGL
        </div>
        <div
          class="meta"
        >
          Alphabet Inc.
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="AMZN"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-positive"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              244.82
            </div>
            <div
              class="label"
            >
              <span
                class="priceup"
              >
                +21.81
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="green arrow up big icon"
            data-testid="ticker-list-item-icon-up"
          />
        </div>
        <div
          class="header"
        >
          AMZN
        </div>
        <div
          class="meta"
        >
          Amazon.com, Inc.
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="ADBE"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-positive"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              275.75
            </div>
            <div
              class="label"
            >
              <span
                class="priceup"
              >
                +11.16
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="green arrow up big icon"
            data-testid="ticker-list-item-icon-up"
          />
        </div>
        <div
          class="header"
        >
          ADBE
        </div>
        <div
          class="meta"
        >
          Adobe Systems Inc.
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="CSCO"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-negative"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              225.81
            </div>
            <div
              class="label"
            >
              <span
                class="pricedown"
              >
                -12.73
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="red arrow down big icon"
            data-testid="ticker-list-item-icon-down"
          />
        </div>
        <div
          class="header"
        >
          CSCO
        </div>
        <div
          class="meta"
        >
          Cisco Systems, Inc.
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="DISCK"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-positive"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              257.54
            </div>
            <div
              class="label"
            >
              <span
                class="priceup"
              >
                +4.77
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="green arrow up big icon"
            data-testid="ticker-list-item-icon-up"
          />
        </div>
        <div
          class="header"
        >
          DISCK
        </div>
        <div
          class="meta"
        >
          Discovery Inc.
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="EBAY"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-negative"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              210.04
            </div>
            <div
              class="label"
            >
              <span
                class="pricedown"
              >
                -29.38
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="red arrow down big icon"
            data-testid="ticker-list-item-icon-down"
          />
        </div>
        <div
          class="header"
        >
          EBAY
        </div>
        <div
          class="meta"
        >
          eBay Inc.
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="NFLX"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-negative"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              230.34
            </div>
            <div
              class="label"
            >
              <span
                class="pricedown"
              >
                -13.58
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="red arrow down big icon"
            data-testid="ticker-list-item-icon-down"
          />
        </div>
        <div
          class="header"
        >
          NFLX
        </div>
        <div
          class="meta"
        >
          Netflix
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="MSFT"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-positive"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              246.52
            </div>
            <div
              class="label"
            >
              <span
                class="priceup"
              >
                +32.02
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="green arrow up big icon"
            data-testid="ticker-list-item-icon-up"
          />
        </div>
        <div
          class="header"
        >
          MSFT
        </div>
        <div
          class="meta"
        >
          Microsoft Corporation
        </div>
      </div>
    </div>
    <div
      class="item ui card tickerListItem"
      data-testid="ticker-list-item"
      id="TSLA"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="glow right floated animating transition"
          data-testid="ticker-list-item-negative"
          style="animation-duration: 2000ms;"
        >
          <div
            class="ui mini statistic"
          >
            <div
              class="value"
            >
              204.50
            </div>
            <div
              class="label"
            >
              <span
                class="pricedown"
              >
                -45.42
              </span>
            </div>
          </div>
          <i
            aria-hidden="true"
            class="red arrow down big icon"
            data-testid="ticker-list-item-icon-down"
          />
        </div>
        <div
          class="header"
        >
          TSLA
        </div>
        <div
          class="meta"
        >
          Tesla Motors, Inc.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
